includes/admin/load.php

```diff
@@ -375,3 +394,35 @@ function addPluginProgressIndicator( message ) {
 		array( 'type' => 'module' )
 	);
 }
+
+/**
+ * Callback function that returns plugin settings link.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $plugin_slug Plugin slug passed to generate the settings link.
+ * @return string|null Either the plugin settings link or null if not available.
+ */
+function perflab_get_plugin_settings_link( string $plugin_slug ): ?string {
```

Comment:
Ah, let's move the WP_HTML_Processor code from above down into here. Then this function can just return the settings _URL_. That will simplify the code in `perflab_plugin_admin_notices()` and move that complexity into this function.